# Patch by Christian Schöniger <csc@fes-aes.de>
#
# http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=522283
#
# If the first physical device of a RAID1 array is faulty or removed, lilo fails to install.
# It doesn't get valid informations about the device (md: RAIDset device 0 = 0x0000) and 
# didn't resume with the next disk.
# In "raid.c" lilo should loop through all devices ( for (pass=0; pass < raid_limit; pass++) )
# to get the disk geometry. But raid_limit is set to md_array_info.active_disks ( 1 ) and
# getting informations about the faulty or removed device is skipped.

The attached patch really scans all devices to get all required information.
diff -Nur lilo-22.8/src/raid.c lilo-22.8/src/raid.c
--- lilo-22.8/src/raid.c	2009-04-01 17:27:06.000000000 +0200
+++ lilo-22.8/src/raid.c	2009-04-01 17:26:48.000000000 +0200
@@ -223,7 +223,7 @@
 	    if (!force_raid) die("Not all RAID-1 disks are active; use '-H' to install to active disks only");
 	    else {
 		warn("Partial RAID-1 install on active disks only; booting is not failsafe\n");
-		raid_limit = md_array_info.active_disks;
+		raid_limit = md_array_info.raid_disks;
 	    }
 	}
 	raid_index = 0;
